addons/udes_stock/tests/test_backorder.py

```diff
@@ -0,0 +1,51 @@
+from . import common
+
+
+class TestBackOrder(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        super(TestBackOrder, cls).setUpClass()
+
+        # Create two quants in location_01
+        cls.create_quant(cls.apple.id, cls.test_location_01.id, 5)
+        cls.create_quant(cls.banana.id, cls.test_location_01.id, 5)
+        # Create a picking for two different products
+        cls.pick1 = cls.create_picking(
+            picking_type=cls.picking_type_pick,
+            products_info=[
+                {"product": cls.apple, "qty": 5},
+                {"product": cls.banana, "qty": 5},
+            ],
+            assign=True,
+        )
+
+    def test_requires_backorder_for_incomplete_moves(self):
+        """
+        Test a picking is flagged as needing a backorder if it has incomplete
+        moves.
+        """
+
+        # Set qty done to 5 for apple and leave as 0 banana
+        apple_move_lines = self.pick1.move_lines.filtered(lambda ml: ml.product_id == self.apple)
```

Comment:
Q: Does this give a warning in the terminal? `qty_done` is on the `stock.move.line` record, not `stock.move`.

We can just do:

`apple_move_line = self.pick1.move_line_ids.filtered(lambda ml: ml.product_id == self.apple)` (Suggest just `apple_move_line` as it is only one record)

This always shows that `_requires_backorder` doesn't check if the move lines that it is given are done... but that is not for this story. :)